src/Tizen.NUI/src/public/WindowSystem/QuickPanelClient.cs

```diff
@@ -0,0 +1,441 @@
+            }
+            remove
+            {
+                int _res;
+                _orientationChanged -= value;
+                if (_orientationChanged == null)
+                {
+                    _res = Interop.QuickPanelClient.DelEventHandler(_tzshQpClient, _orientationEventHandler);
+                    _tzsh.ErrorCodeThrow(_res);
+                }
+            }
+        }
+
+        /// <summary>
+        /// Enumeration for type of quickpanel service window.
+        /// </summary>
+        /// <since_tizen> 8 </since_tizen>
+        public enum QuickPanelType
+        {
+            /// <summary>
+            /// Unknown type. There is no quickpanel service.
+            /// </summary>
+            Unknown = 0x0,
+            /// <summary>
+            /// System default type.
+            /// </summary>
+            SystemDefault = 0x1,
+            /// <summary>
+            /// Context menu type.
+            /// </summary>
+            ContextMenu = 0x2,
+            /// <summary>
+            /// Apps menu type.
+            /// </summary>
+            AppsMenu = 0x3,
+        }
+
+        /// <summary>
+        /// Enumeration for visible state of quickpanel service window.
+        /// </summary>
+        /// <since_tizen> 8 </since_tizen>
+        public enum StateVisible
+        {
+            /// <summary>
+            /// Unknown state. There is no quickpanel service.
+            /// </summary>
+            /// <since_tizen> 8 </since_tizen>
+            Unknown = 0x0,
+            /// <summary>
+            /// Shown state.
+            /// </summary>
+            /// <since_tizen> 8 </since_tizen>
+            Shown = 0x1,
+            /// <summary>
+            /// Hidden state.
+            /// </summary>
+            /// <since_tizen> 8 </since_tizen>
+            Hidden = 0x2,
+        }
+
+        /// <summary>
+        /// Enumeration for scrollable state of quickpanel service window.
+        /// </summary>
+        /// <since_tizen> 8 </since_tizen>
+        public enum StateScrollable
+        {
+            /// <summary>
+            /// Unknown state. There is no quickpanel service.
+            /// </summary>
+            /// <since_tizen> 8 </since_tizen>
+            Unknown = 0x0,
+            /// <summary>
+            /// Scrollable state.
+            /// </summary>
+            /// <since_tizen> 8 </since_tizen>
+            Set = 0x1,
+            /// <summary>
+            /// Not scrollable state.
+            /// </summary>
+            /// <since_tizen> 8 </since_tizen>
+            Unset = 0x2,
+            /// <summary>
+            /// Retain scrollable state.
+            /// </summary>
+            /// <since_tizen> 8 </since_tizen>
+            Retain = 0x3,
+        }
+
+        /// <summary>
+        /// Enumeration for orientation state of quickpanel service window.
+        /// </summary>
+        /// <since_tizen> 8 </since_tizen>
+        public enum StateOrientation
```

Comment:
NUI define WindowOrientation. Please check if you can reuse that.
 If you need to define your own, Orientation, WindowOrientation or QuickPanelOrientation is better.

Reply:
I changed this to WindowOrientation.
But we need to talk about specific definition of inverse orientation